### PR TITLE
Move RcuVector and relevant support classes to vespalib

### PR DESCRIPTION
**Pull request by vekterli**
@geirst @toregge @baldersheim please review

Having RCU support available across all our C++ modules open up
new opportunities for optimizations. Also a prerequisite for moving
B-tree/datastore code to vespalib.

This changes the following:
- `RcuVector` moved from `searchlib` to `vespalib`
- `MemoryUsage` moved from `searchlib` to `vespalib`
- Introduce a simplified, more generic `GrowStrategy` in `vespalib`
  used by the moved `RcuVector` which does not have any notion of
  documents. Existing searchlib `GrowStrategy` gets a simple function
  to convert to this generic strategy. I've replaced instances of the searchlib
  strategy with the one in vespalib where these have only been used as
  inputs to RCU vectors. It's quite possible that more could be replaced.

### CODE REVIEW

**Review by baldersheim — APPROVED** [review]

File: vespalib/src/vespa/vespalib/util/growstrategy.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+#pragma once
+
+#include <cstdint>
+
+namespace vespalib {
+
+class GrowStrategy {
+private:
+    uint32_t _initialCapacity;
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/util/growstrategy.h]: Consider changing all use of uint32_t to size_t to match what is used in rcuvector.

**Review by geirst — APPROVED** [review]
🏅 Only a nitpick to consider.

File: vespalib/src/vespa/vespalib/util/growstrategy.h (hunk 1 above)
Comment (vekterli) [reply vespalib/src/vespa/vespalib/util/growstrategy.h]: Done

**Review by geirst — APPROVED** [review]
👍

**Review by toregge — APPROVED** [review]

**Pull request merged**